Description: Switch the Makefile to be multiarch-friendly
 google-authenticator looks for /usr/lib/libdl.so to determine whether
 or not to link -ldl, but since the multiarch transition process moved
 libdl to a different location, that test no longer finds it, leading
 to a FTBFS on multiarch-enabled systems.
 .
 Work around that by just always linking libdl, which we definitely
 always want to do with glibc.
Author: Evan Broder <evan@ebroder.net>
Origin: vendor
Bug: http://code.google.com/p/google-authenticator/issues/detail?id=74

Index: ubuntu/libpam/Makefile
===================================================================
--- ubuntu.orig/libpam/Makefile	2011-05-31 11:40:31.878079519 -0700
+++ ubuntu/libpam/Makefile	2011-05-31 11:52:43.952386500 -0700
@@ -47,18 +47,17 @@
 	               pam_google_authenticator_unittest
 
 google-authenticator: google-authenticator.o base32.o hmac.o sha1.o
-	$(CC) -g $(LDFLAGS) $(shell [ -f /usr/lib/libdl.so ] && echo " -ldl") \
-	      -o $@ $+
+	$(CC) -g $(LDFLAGS)                                                   \
+	      -o $@ $+ -ldl
 
 demo: demo.o pam_google_authenticator_demo.o base32.o hmac.o sha1.o
 	$(CC) -g $(LDFLAGS) -rdynamic                                         \
-	      $(shell [ -f /usr/lib/libdl.so ] && echo " -ldl") -o $@ $+
+	      -o $@ $+ -ldl
 
 pam_google_authenticator_unittest: pam_google_authenticator_unittest.o        \
                                    base32.o hmac.o sha1.o
 	$(CC) -g $(LDFLAGS) -rdynamic -lc                                     \
-              $(shell [ -f /usr/lib/libdl.so ] && echo " -ldl")               \
-              -o $@ $+
+              -o $@ $+ -ldl
 
 pam_google_authenticator.so: base32.o hmac.o sha1.o
 pam_google_authenticator_testing.so: base32.o hmac.o sha1.o
